src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1176,10 +1179,16 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(SyntaxNode t
             Debug.Assert(!typeArguments.IsEmpty);
             type = type.Construct(typeArguments);
 
-            if (ShouldCheckConstraints)
+            if (ShouldCheckConstraints && ConstraintsHelper.RequiresChecking(type))
             {
                 bool includeNullability = Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking);
-                type.CheckConstraintsForNonTuple(this.Conversions.WithNullability(includeNullability), typeSyntax, typeArgumentsSyntax, this.Compilation, basesBeingResolved, diagnostics);
+                var conversions = this.Conversions.WithNullability(includeNullability);
+                if (includeNullability && Flags.Includes(BinderFlags.SuppressNullableContraintChecks))
+                {
+                    diagnostics.Add(new LazyNullableContraintChecksDiagnosticInfo(type, conversions, this.Compilation), typeSyntax.GetLocation());
+                    conversions = conversions.WithNullability(includeNullability: false);
```

Comment:
>conversions = conversions.WithNullability(includeNullability: false); [](start = 20, length = 69)

It makes sense that postpone some portion of the check, and so we switch the mode on conversions to perform the rest of the constraints.
I would have expected a similar pattern in `BindNamespaceOrTypeOrAliasSymbol` which you modified above (line 375). Why is it different there?

Reply:
Good catch. Fixed.

---
In reply to: [209368024](https://github.com/dotnet/roslyn/pull/29197#discussion_r209368024) [](ancestors = 209368024)